app/main.py

```diff
@@ -1,3 +1,49 @@
+def _convert_to_km(other: (int, float)) -> float:
+    if isinstance(other, Distance):
+        return other.km
+    elif isinstance(other, (int, float)):
+        return float(other)
+    raise TypeError("Unsupported type for arithmetic operation")
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float = 0.0) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: (int, float)) -> "Distance":
```

Comment:
1. Use annotations from `__future__` module so you can use class name as a type hint inside of the class
2. Typing of (int, float) is misleading because it implies that other should be a tuple of int and float, correct way is to use Union[int, float] or use pipe int | float.
3. other can also be of Distance type.
4. Please check and fix in all methods where applicable